scipy/ndimage/tests/test_ndimage.py

```diff
@@ -4578,3 +4578,16 @@ def test_dilation_scalar_size(self):
         result = ndimage.grey_dilation(self.array, size=3)
         assert_array_almost_equal(result, self.dilated3x3)
 
+class TestBinaryOpening:
+
+    def setup_method(self):
+        a = numpy.zeros((5,5), dtype=int)
```

Comment:
The natural input of `binary_*` operations are boolean arrays. I'd suggest making this `dtype=bool`, which I'm guessing would probably also spare you the `.astype(int)` below?